orttraining/orttraining/python/training/optim/lr_scheduler.py

```diff
@@ -0,0 +1,290 @@
+import math
+
+from . import config
+
+
+class LRScheduler(object):
+    r"""Base class for implementing custom learning rate schedulers
+
+    Schedulers can be either statefull or stateless.
+    Stateless implementations can only rely on information available at
+    :py:class:`.TrainStepInfo` and :py:attr:`.optimizer_config`.
+    Statefull, on the other hand, can store more parameters by overriding
+    the constructor.
+
+    In both cases, once the scheduler is configured, no user code is needed
+    to update learning rate during each train step.
+
+    NOTE: This class should never be instantiated, but used as an abstract class.
+
+    NOTE: Current implementation doesn't support 'lr' within :py:attr:`param_groups` entries.
+
+    Args:
+        optimizer_config (optim._OptimizerConfig): optimizer config.
+            One of :py:class:`.optim.Adam`, :py:class:`.optim.Lamb` or :py:class:`.optim.SGD`.
+    """
+
+    def __init__(self, optimizer_config):
+        assert isinstance(optimizer_config, config._OptimizerConfig),\
+            "optimizer_config must be :py:class:`.optim._OptimizerConfig"
+        for group in optimizer_config.param_groups:
+            assert 'lr' not in group, "'lr' is not supported inside param_groups"
+        self.optimizer_config = optimizer_config
+        self._last_lr = []
+
+    def get_lr(self, train_step_info):
+        r"""Returns a list of learning rate
+
+        Args:
+            train_step_info (:py:class:`.TrainStepInfo`): runtime info for current training step
+
+        Returns:
+            ordered :py:obj:`list` of learning rates.
+                The first entry is the default learning rate and
+                    the remaining, refer to each :py:class:`optim._OptimizerConfig`s parameter group.
+            NOTE: Currently, only the default learning rate is supported,
```

Comment:
Checking if we have plans to do anything different in the future?

Reply:
After talking to Sherlock and Liqun, we agreed this feature is possible through some changes on ORT backend and it might be requested by users. Therefore, the new frontend does support multiple lr to be specified, but we are blocking it until backend provides support